### PR TITLE
spanner-jdbc-cleanup - Step 05 - Add missing test cases

### PR DESCRIPTION
**Pull request by olavloite**
Adds miscellaneous unit tests that had not yet been merged in.
Two test cases in the `JdbcAbortedTransactionTest` will fail until #5952 has been merged.

### CODE REVIEW

**Comment by kolea2** [comment]
Will review after rebase :)

**Comment by olavloite** [comment]
> Will review after rebase :)

Done rebasing, and I cherry-picked the fix for #5951 into this PR in order to prevent the `JdbcAbortedTransactionTest` to fail.

File: google-cloud-clients/google-cloud-contrib/google-cloud-spanner-jdbc/src/test/java/com/google/cloud/spanner/jdbc/ConnectionStatementWithNoParametersTest.java [hunk 1]
```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2018 Google LLC
```
Comment (kolea2) [inline google-cloud-clients/google-cloud-contrib/google-cloud-spanner-jdbc/src/test/java/com/google/cloud/spanner/jdbc/ConnectionStatementWithNoParametersTest.java]: 2019

File: google-cloud-clients/google-cloud-contrib/google-cloud-spanner-jdbc/src/test/java/com/google/cloud/spanner/jdbc/DdlClientTest.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2018 Google LLC
```
Comment (kolea2) [inline google-cloud-clients/google-cloud-contrib/google-cloud-spanner-jdbc/src/test/java/com/google/cloud/spanner/jdbc/DdlClientTest.java]: 2019

File: google-cloud-clients/google-cloud-contrib/google-cloud-spanner-jdbc/src/test/java/com/google/cloud/spanner/jdbc/ReadOnlyStalenessTest.java [hunk 3]
```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright 2018 Google LLC
```
Comment (kolea2) [inline google-cloud-clients/google-cloud-contrib/google-cloud-spanner-jdbc/src/test/java/com/google/cloud/spanner/jdbc/ReadOnlyStalenessTest.java]: 2019

File: google-cloud-clients/google-cloud-contrib/google-cloud-spanner-jdbc/src/test/java/com/google/cloud/spanner/jdbc/SpannerJdbcExceptionMatcher.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2018 Google LLC
```
Comment (kolea2) [inline google-cloud-clients/google-cloud-contrib/google-cloud-spanner-jdbc/src/test/java/com/google/cloud/spanner/jdbc/SpannerJdbcExceptionMatcher.java]: 2019

File: google-cloud-clients/google-cloud-contrib/google-cloud-spanner-jdbc/src/test/java/com/google/cloud/spanner/jdbc/StatementTimeoutTest.java [hunk 5]
```diff
@@ -0,0 +1,961 @@
+/*
+ * Copyright 2018 Google LLC
```
Comment (kolea2) [inline google-cloud-clients/google-cloud-contrib/google-cloud-spanner-jdbc/src/test/java/com/google/cloud/spanner/jdbc/StatementTimeoutTest.java]: 2019

File: google-cloud-clients/google-cloud-contrib/google-cloud-spanner-jdbc/src/test/java/com/google/cloud/spanner/stresstest/DatabaseClientStressTest.java [hunk 6]
```diff
@@ -0,0 +1,548 @@
+/*
```
Comment (kolea2) [inline google-cloud-clients/google-cloud-contrib/google-cloud-spanner-jdbc/src/test/java/com/google/cloud/spanner/stresstest/DatabaseClientStressTest.java]: This is great, but I don't think this is the appropriate place to put this type of test alongside other JUnit tests. For example, we have a different module for our conformance tests - not sure if we should think about a new module for stress testing.

That said, can we move this to a future PR after we think about it some more?

**Comment by olavloite** [comment]
Removed stress tests and updated copyright years.

Note that the stress tests were not executed alongside with the unit tests, as they were added to a test group that was excluded from normal test execution. If we want to add these back to the library, we should also add this exclusion again to prevent test execution from taking too much time.

**Pull request merged**